src/recommender.py

```diff
@@ -96,22 +98,28 @@ class GraphDB:
     """Graph database interface."""
 
     @staticmethod
-    def execute_gremlin_dsl(payload):
+    def execute_gremlin_dsl(payload, unit_test=False):
         """Execute the gremlin query and return the response."""
         try:
-            response = get_session_retry().post(GREMLIN_SERVER_URL_REST, data=json.dumps(payload))
+            if unit_test:
+                url = 'http://bayesian-gremlin-http-preview-b6ff-bayesian-preview.b6ff.' \
+                      'rh-idev.openshiftapps.com/'
+                graph_req = get_session_retry().post(url, data=json.dumps(payload))
```

Comment:
I'd suggest to simply use `BAYESIAN_GREMLIN_HTTP_SERVICE_HOST` and `BAYESIAN_GREMLIN_HTTP_SERVICE_PORT` environment variables, it it's possible (probably via new Dockerfile?)

Reply:
and we can mock the get_session_retry().post() call by mock (this is the harder way)